src/log.ts

```diff
@@ -58,19 +57,103 @@ export class ConsoleHandler extends BaseHandler {
   }
 }
 
+class FileHandler extends BaseHandler {
+  format(record: LogRecord): string {
+    let msg = record.datetime.toISOString() + " ";
+
+    switch (record.level) {
+      case LogLevels.DEBUG:
+        msg += "[DEBUG]   ";
+        break;
+      case LogLevels.INFO:
+        msg += "[INFO]    ";
+        break;
+      case LogLevels.WARNING:
+        msg += "[WARNING] ";
+        break;
+      case LogLevels.ERROR:
+        msg += "[ERROR]   ";
+        break;
+      case LogLevels.CRITICAL:
+        msg += "[CRITICAL]";
+        break;
+      default:
+        break;
+    }
+
+    msg += ` ${stripANSII(record.msg)}`;
+
+    for (const arg of record.args) {
+      msg += ` ${stripANSII(Deno.inspect(arg))}`;
+    }
+
+    return msg;
+  }
+
+  log(msg: string): void {
+    masterLogRecord += msg + "\n";
+  }
+}
+
 /** Setup custom deno logger. Follows format:
  * `[LEVEL] <msg> <args>` */
-export async function setupLog(): Promise<void> {
-  const level = DEBUG_LEVEL; // TODO(@qu4k): make it an flag
+export async function setupLog(
+  debugEnabled = false,
+): Promise<void> {
+  detailedLog = debugEnabled;
   await log.setup({
     handlers: {
-      [DEFAULT_HANDLER]: new ConsoleHandler(level),
+      console: new ConsoleHandler(debugEnabled ? "DEBUG" : "INFO"),
+      file: new FileHandler("DEBUG"),
     },
     loggers: {
       default: {
-        level,
-        handlers: [DEFAULT_HANDLER],
+        level: "DEBUG",
+        handlers: ["console", "file"],
       },
     },
   });
 }
+
+export async function writeLogFile() {
+  const encoder = new TextEncoder();
+
+  const args = `Arguments:\n  ${Deno.args}\n\n`;
+  const denoVersion =
+    `Deno version:\n  deno: ${Deno.version.deno}\n  v8: ${Deno.version.v8}\n  typescript: ${Deno.version.typescript}\n\n`;
+  const eggsVersion = `Eggs version:\n  ${version}\n\n`;
+  const platform = `Platform:\n  ${Deno.build.target}\n\n`;
+
+  await Deno.writeFile(
+    logFile,
+    encoder.encode(
+      args +
+        denoVersion +
+        eggsVersion +
+        platform +
+        masterLogRecord,
+    ),
+  );
+}
+
+export async function handleError(err: Error) {
+  log.critical(`An unexpected error occurred: "${err.message}"`, err.stack);
+  await writeLogFile();
+  log.info(
+    `If you think this is a bug, please open a bug report with the information provided in ${
+      bold(resolve(Deno.cwd(), logFile))
+    }.`,
+  );
+  log.info(
+    `Visit ${
+      bold("https://docs.nest.land/eggs/")
```

Comment:
If possible, stylise links to use underline and/or bolding. I'll leave the "and/or" part up to you depending on what feels right.

Reply:
@zorbyte What do you mean here?  It's already in bold.

Reply:
@oganexon Something like [this](https://github.com/sindresorhus/terminal-link) maybe?

Reply:
I'm going to try  
Even though my terminal's already doing it for me 😋

Reply:
😝

Reply:
Underline + Bold, try using either or both until it looks good